x-pack/plugins/apm/server/routes/fleet/apm_java_agent_versions.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import fetch from 'node-fetch';
+
+const mavenJavaAgentUrl =
+  'https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent/maven-metadata.xml';
+
+const versionRegex = /<version>(\d+\.\d+\.\d+.*?)<\/version>/gm;
+const latestVersionRegex = /<latest>(\d+\.\d+\.\d+.*?)<\/latest>/;
+
+export async function getApmJavaAgentVersionsFromRegistry() {
+  const response = await (await fetch(mavenJavaAgentUrl)).text();
+  const versions = [...response.matchAll(versionRegex)].map(
+    (aMatch) => aMatch[1]
+  );
+  const latestVersion = response.match(latestVersionRegex)![1];
+  return {
+    versions,
```

Comment:
Reverse the array so that the newer versions are on the top of the dropdown.

Reply:
See also https://github.com/elastic/kibana/issues/133374#issuecomment-1148816768